frontend/test/src/container/Header.spec.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+import { expect } from 'chai'
+import { Header as HeaderWithoutHOC } from '../../../src/container/Header.jsx'
+import sinon from 'sinon'
+import { user, userFromApi } from '../../hocMock/redux/user/user.js'
+import { appList } from '../../hocMock/redux/appList/appList.js'
+import {
+  SET,
+  UPDATE,
+  USER,
+  USER_DISCONNECTED
+} from '../../../src/action-creator.sync.js'
+import { FETCH_CONFIG } from '../../../src/util/helper.js'
+import {
+  restoreHistoryCallBack,
+  isFunction
+} from '../../hocMock/helper'
+import { shallow } from 'enzyme'
+import { mockPostUserLogout204 } from '../../apiMock'
+
+describe('In <Header />', () => {
+  const setUserDisconnectedCallBack = sinon.spy()
+  const setUserLangCallBack = sinon.spy()
+  const updateUserCallBack = sinon.spy()
+
+  const dispatchMock = (params) => {
+    if (isFunction(params)) return params(dispatchMock)
+
+    const { type } = params
+    switch (type) {
+      case `${SET}/${USER_DISCONNECTED}`: setUserDisconnectedCallBack(); break
+      case `${SET}/${USER}/Lang`: setUserLangCallBack(); break
+      case `${UPDATE}/${USER}`: updateUserCallBack(); break
+    }
+    return params
+  }
+
+  const props = {
+    user: user,
+    appList: appList,
+    searchResult: {},
+    tlmManager: {
+      closeLiveMessageConnection: () => {}
+    },
+    lang: [],
+    system: {
+      config: {
+        email_notification_activated: true
+      }
+    },
+    location: {
+      pathname: {
+        startsWith: () => {}
```

Comment:
startsWith() is a function of the String prototype, you shouldn't have to mock it as long as it is used on a string

Reply:
Changed